book/src/configuration.md

```diff
@@ -64,6 +64,7 @@ Its settings will be merged with the configuration directory `config.toml` and t
 | `text-width` | Maximum line length. Used for the `:reflow` command and soft-wrapping if `soft-wrap.wrap-at-text-width` is set | `80` |
 | `workspace-lsp-roots` | Directories relative to the workspace root that are treated as LSP roots. Should only be set in `.helix/config.toml` | `[]` |
 | `default-line-ending` | The line ending to use for new documents. Can be `native`, `lf`, `crlf`, `ff`, `cr` or `nel`. `native` uses the platform's native line ending (`crlf` on Windows, otherwise `lf`). | `native` |
+| `insert-final-newline` | Whether to automatically insert a trailing line-ending on write if missing | `false` |
```

Comment:
I agree :+1: 

This is the default behaviour for almost all shell tools and I would personally enable it (I get random diffs with helix all the time because I accidentally delete a trailing newline somewhere). There are definitely cases where its useful to disable this but they are probably more niche.

Reply:
> I get random diffs with helix all the time because I accidentally delete a trailing newline somewhere

Me too, it needs to stop, please consider adding this to 23.09 https://github.com/helix-editor/helix/milestone/6

Reply:
Default changed. There were some failing tests as a result; those have been fixed.